fuse/plugins/detector_physics/electron_drift.py

```diff
@@ -327,18 +333,35 @@ def get_s2_drift_time_params(self, xy_int, z_int):
         Returns:
             returns two arrays of floats (mean drift time, drift time spread)
         """
-        drift_velocity_liquid = self.get_avg_drift_velocity(z_int, xy_int)
+        drift_velocity_below_gate = self.get_avg_drift_velocity(z_int, xy_int)
+        drift_velocity_above_gate = self.liquid_level / self.drift_time_gate
         if self.enable_diffusion_longitudinal_map:
             diffusion_constant_longitudinal = self.diffusion_longitudinal_map(
                 z_int, xy_int
             )  # cm²/ns
         else:
             diffusion_constant_longitudinal = self.diffusion_constant_longitudinal
 
-        drift_time_mean = -z_int / drift_velocity_liquid + self.drift_time_gate
+        drift_time_below_gate = -z_int / drift_velocity_below_gate
+        drift_time_above_gate = self.drift_time_gate
+
+        drift_time_mean = drift_time_below_gate + drift_time_above_gate
         drift_time_mean = np.clip(drift_time_mean, 0, np.inf)
-        drift_time_spread = np.sqrt(2 * diffusion_constant_longitudinal * drift_time_mean)
-        drift_time_spread /= drift_velocity_liquid
+
+        drift_time_spread2 = (
+            2
+            * diffusion_constant_longitudinal
+            * drift_time_below_gate
+            / drift_velocity_below_gate**2
+        )
+        drift_time_spread2 += (
+            2
+            * diffusion_constant_longitudinal
+            * drift_time_above_gate
+            / drift_velocity_above_gate**2
+        )
+        drift_time_spread = np.sqrt(drift_time_spread2)
```

Comment:
Here I would like to use a little more meaningful names for the two drift time spread values. Maybe `drift_time_spread_above_gate_squared` and `drift_time_spread_below_gate_squared` can work and then `drift_time_spread = np.sqrt(drift_time_spread_above_gate_squared + drift_time_spread_below_gate_squared)`?

Reply:
Sure, modified